### ISSUE TITLE
contract - define the column types in yml file - dbt-clickhouse==1.4.8

### ISSUE THREAD

**Issue by zli06160** [ClickHouse/dbt-clickhouse#189]
Hello, if I understand well, the feature [**contract**](https://docs.getdbt.com/reference/resource-configs/contract) is not yet supported in dbt-clickhouse. 

**Is your feature request related to a problem? Please describe.**

- In https://github.com/ClickHouse/dbt-clickhouse/issues/149 the feature `contract` is mentioned
- I tried with the `table` and `distributed_table` materializations, it seems that this feature is not yet supported.

**Describe the solution you'd like**

- I want to define the column types in the `model/xxx.yml` yaml file (rather than in each .sql file), like the following (example from official doc: https://docs.getdbt.com/reference/resource-configs/contract)

`model/xxx.yml`:
````yaml
models:
  - name: dim_customers
    config:
      materialized: table
      contract:
        enforced: true
    columns:
      - name: customer_id
        data_type: int
        constraints:
          - type: not_null
      - name: customer_name
        data_type: string
````

**Describe alternatives you've considered**

- I want to know if this feature is under development, if so, when will be the expected release?

**Additional context**

dbt-clickhouse==1.4.8 (not 1.5.0 yet)

**Comment by zli06160** [comment]
@dataders @gfunc @genzgd Thanks in advance :)

**Comment by genzgd** [comment]
This is not currently under development.  We don't yet have a timeline for dbt 1.5.0 features.

**Comment by zli06160** [comment]
@genzgd Thank you for the reply.

**Comment by genzgd** [comment]
Support for this was added in the recently released 1.5.x versions -- however, for the foreseeable future contract support expects explicit ClickHouse type names (e.g., `Int32` or `UInt8` instead of just `int`), since trying to determine functional equivalence between the various available ClickHouse data types is not currently practicable.
